app/components/selective_button/selective_button.props.ts

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+
+import {
+  StyleProp,
+  TextStyle,
+  TouchableOpacityProps,
+  ViewStyle,
+} from 'react-native';
+
+export interface SelectiveButtonProps extends TouchableOpacityProps {
+  /**
+   * Text which is looked up via i18n.
+   */
+  /**
+   * store value of the button
+   */
+  state?: boolean;
+  /**
+   * The text to display if not using `tx` or nested components.
+   */
+  text?: string;
+  /**
+   * The image to display as icon
+   */
+  icon?: string;
```

Comment:
icon is necessary. And this is imagesourproptype
```suggestion
  icon: ImageSourcePropType;
```